cpp/src/arrow/compare.cc

```diff
@@ -1522,5 +1524,54 @@ bool TypeEquals(const DataType& left, const DataType& right, bool check_metadata
     return visitor.result();
   }
 }
+namespace {
+
+using ValueType = ArrayStatistics::ValueType;
+
+bool DoubleEquals(const double& left, const double& right, const EqualOptions& options) {
+  bool result;
+  auto visitor = [&](auto&& compare_func) { result = compare_func(left, right); };
+  VisitFloatingEquality<double>(options, options.use_atol(), std::move(visitor));
+  return result;
+}
 
+bool ArrayStatisticsValueTypeEquals(const std::optional<ValueType>& left,
+                                    const std::optional<ValueType>& right,
```

Comment:
`ValueType` is only used here.
Can we remove `using ValueType = ArrayStatistics::ValueType;`?